### PR TITLE
Remove uses of NodeInfoMap outside of snapshot and cache

### PR DESCRIPTION
**Pull request by alculquicondor**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

NodeInfoMap is an internal data structure to the snapshot and it can sometimes contain nodes that are not fully initialized.

The snapshot's listers.NodeInfoLister implementation already handles this. All plugins, extensions and the generic scheduler should use it instead of the map.

**Which issue(s) this PR fixes**:

Refs #86920

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by Huang-Wei — COMMENTED** [review]
LGTM generally. Some nits below.

File: pkg/scheduler/core/extender.go [hunk 1]
```diff
@@ -210,46 +211,43 @@ func (h *HTTPExtender) ProcessPreemption(
...
 ) (map[*v1.Node]*extenderv1.Victims, error) {
 	nodeToVictims := map[*v1.Node]*extenderv1.Victims{}
 	for nodeName, metaVictims := range nodeNameToMetaVictims {
 		victims := &extenderv1.Victims{
 			Pods: []*v1.Pod{},
 		}
+		nodeInfo, err := nodeInfos.Get(nodeName)
+		if err != nil {
+			return nil, err
+		}
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply pkg/scheduler/core/extender.go]: Done

File: pkg/scheduler/core/extender.go [hunk 2]
```diff
@@ -210,46 +211,43 @@ func (h *HTTPExtender) ProcessPreemption(
...
 			}
 			victims.Pods = append(victims.Pods, pod)
 		}
-		nodeToVictims[nodeNameToInfo[nodeName].Node()] = victims
+		nodeToVictims[nodeInfo.Node()] = victims
 	}
 	return nodeToVictims, nil
 }
 
 // convertPodUIDToPod returns v1.Pod object for given MetaPod and node name.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply pkg/scheduler/core/extender.go]: Done

**Pull request merged**